src/main/java/io/jenkins/plugins/sample/OnboardingTaskBuilder.java

```diff
@@ -0,0 +1,257 @@
+import hudson.EnvVars;
+import hudson.Extension;
+import hudson.FilePath;
+import hudson.Launcher;
+import hudson.model.AbstractProject;
+import hudson.model.Run;
+import hudson.model.TaskListener;
+import hudson.tasks.BuildStepDescriptor;
+import hudson.tasks.Builder;
+import hudson.util.FormValidation;
+import hudson.util.ListBoxModel;
+import jenkins.model.GlobalConfiguration;
+import jenkins.tasks.SimpleBuildStep;
+import org.jenkinsci.Symbol;
+import org.kohsuke.stapler.DataBoundConstructor;
+import org.kohsuke.stapler.DataBoundSetter;
+import org.kohsuke.stapler.QueryParameter;
+
+import javax.servlet.ServletException;
+import java.io.*;
+import java.lang.reflect.Type;
+import java.time.Instant;
+import java.util.*;
+import java.util.stream.Collectors;
+
+public class OnboardingTaskBuilder extends Builder implements SimpleBuildStep {
+
+    private static final String CATEGORY_JOBS_FILE = "category_jobs.json";
+    private static final Gson gson = new GsonBuilder().registerTypeAdapter(Instant.class, new InstantAdapter())
+            .setPrettyPrinting().create();
+    private static final Object lock = new Object();
+
+    public static class InstantAdapter extends TypeAdapter<Instant> {
+        @Override
+        public void write(JsonWriter out, Instant value) throws IOException {
+            out.value(value.toString());
+        }
+
+        @Override
+        public Instant read(JsonReader in) throws IOException {
+            return Instant.parse(in.nextString());
+        }
+    }
+
+    private final String name;
+    private String selectedCategory;
+    @DataBoundConstructor
+    public OnboardingTaskBuilder(String name) {
+        this.name = name;
+    }
+
+    public String getName() {
+        return name;
+    }
+
+    @DataBoundSetter
+    public void setSelectedCategory(String category) {
+         this.selectedCategory = category;
+    }
+
+    public String getSelectedCategory() {
+        return selectedCategory;
+    }
+
+    @Override
+    public void perform(Run<?, ?> run, FilePath workspace, EnvVars env, Launcher launcher, TaskListener listener)
+            throws InterruptedException, IOException {
+        OnboardingSectionConfiguration onboardingSectionConfig = GlobalConfiguration.all()
+                .get(OnboardingSectionConfiguration.class);
+        if (Objects.nonNull(onboardingSectionConfig)) {
+            Optional<CategoryEntry> categoryEntryOptional = onboardingSectionConfig.getCategoryConfig().getCategories()
+                    .stream()
+                    .filter(category -> category.getUuid().equals(selectedCategory)).findFirst();
+            listener.getLogger().println("I'm " + name + (categoryEntryOptional
+                    .map(categoryEntry -> " and I have selected " + categoryEntry.getName()).orElse("")));
+            categoryEntryOptional.ifPresent(categoryEntry -> saveJobNamePerCategory(run.getFullDisplayName(), categoryEntry.getName()));
```

Comment:
Avoid using `run.getFullDisplayName` or any display name-related methods for non-user-facing IDs. Use `run.getExternalizableId` if you need the build number, or `run.getParent().getFullName()` if you only care about the job. If you need the display name later, first look up the job/build using the full name/externalizable ID.